### PR TITLE
Add zvector support for torchinductor

### PR DESCRIPTION
**Pull request by tehbone**
This updates the torchinductor C++/OpenMP backend to enable the use of vectorized operations when compiled for the s390x arch.

I also noticed that `vec256_zarch.h` was not getting installed, so this PR also addresses that. Finally, the `VecISA` arch flags were not being propagated to the compilation no matter the platform. This prevented torchinductor from utilizing the zvector instructions, period. I ran a simple compile test on x86 to confirm that the addition of the flags to the AVX2 and AVX512 cases do not break compilation.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -555,7 +567,7 @@ def __bool__(self):
 
 invalid_vec_isa = InvalidVecISA()
 supported_vec_isa_list = [VecAVX512(), VecAVX2()]
```
Comment (jgong5) [inline torch/_inductor/codecache.py:569]: consider rename it to `supported_x86_vec_isa_list` to be more accurate?

**Review by jgong5 — APPROVED** [review]
LGTM.

File: torch/_inductor/codecache.py (hunk 1 above)
Comment (tehbone) [reply torch/_inductor/codecache.py:569]: Yes, I can do that.

Comment (jgong5) [reply torch/_inductor/codecache.py:569]: Seems quite a few CI failures. Please check.

Comment (tehbone) [reply torch/_inductor/codecache.py:569]: Yes, I missed updates in the test path and in cases involving other paths of invocation. Reworking.

**Comment by tehbone** [comment]
If I separate the part of this PR which at least ships the vector headers, could that make the 2.1.0 build?

**Pull request closed without merging**